api/api/utils/search_context.py

```diff
@@ -41,7 +44,12 @@ def build(cls, results: list[Hit], origin_index: OriginIndex) -> Self:
             # cf: https://github.com/WordPress/openverse/issues/2154
             Q(
                 "terms",
-                **{"identifier.keyword": [result.identifier for result in results]},
+                **{"identifier.keyword": all_result_identifiers},
+            )
+            if len(results) > 1
+            else Q(
+                "term",
+                **{"identifier.keyword": all_result_identifiers[0]},
             )
```

Comment:
This is fine, but FWIW I don't think it's entirely necessary. `terms` should still work even with a list of 1 and I'd be surprised if there is a significant performance difference between the two. Have you already measured that? The only reason I'd suggest not doing this is (a) it feels a little like pre-optimising and (b) it makes the code slightly more obscure.